homeassistant/components/withings/__init__.py

```diff
@@ -336,14 +343,15 @@ async def async_webhook_handler(
                 "Parameter appli not provided", message_code=20
             )
 
-        try:
-            appli = NotifyAppli(int(params.getone("appli")))  # type: ignore[arg-type]
-        except ValueError:
-            return json_message_response("Invalid appli provided", message_code=21)
+        notification_category = to_enum(
+            NotificationCategory,
+            int(params.getone("appli")),  # type: ignore[arg-type]
+            NotificationCategory.UNKNOWN,
+        )
 
         for coordinator in coordinators.values():
```

Comment:
The linear search here could probably be avoided by indexing the categories in the future with a dict that points to the coordinators for each category

Reply:
I think this would work well with the dataclass